src/main/java/sg/christopher/duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+
+        if (foundTasks.size() == 0) {
+            System.out.println("No task matching your search term was found. Perhaps try another search term?");
+            return;
+        }
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < foundTasks.size(); ++i) {
+            Task task = foundTasks.get(i);
+            System.out.println(i + 1 + ". " + task);
+        }
+    }
+
+    private static void lsHandler() {
+        if (savedItems.size() == 0) {
+            System.out.println("No tasks found. Start adding your first few tasks!");
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < savedItems.size(); ++i) {
+            Task task = loadItem(i);
+            System.out.println(i + 1 + ". " + task);
+        }
+        printRemainingCount();
+    }
+
+    private static void deleteHandler(String userInput) {
+        int taskNo;
+        try {
+            taskNo = Integer.parseInt(userInput.split(" ")[1]);
+        } catch (ArrayIndexOutOfBoundsException aioobe) {
+            System.out.println("ERROR: No task no. found. Did you input the task no. of the task you'd like to delete?");
+            return;
+        } catch (NumberFormatException nfe) {
+            System.out.println("ERROR: Unrecognized task. Please input the task no. of the task you'd like to delete.");
+            return;
+        }
+
+        Task task;
+        try {
+            task = savedItems.remove(taskNo - 1);
+        } catch (IndexOutOfBoundsException ioobe) {
+            System.out.println("ERROR: Task no. not found. Does that task exist?");
+            return;
+        }
+
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(taskNo + ". " + task);
+        printRemainingCount();
+    }
+
+    private static void printRemainingCount() {
+        System.out.println("You now have " + savedItems.size() + " tasks in the list.");
+    }
+
+    private static CommandType getCommandType(String command) {
+        switch (command) {
+        case "bye":
+            // Fallthrough
+        case "exit":
+            return CommandType.EXIT;
+        case "todo":
+            return CommandType.TODO;
+        case "deadline":
+            return CommandType.DEADLINE;
+        case "event":
+            return CommandType.EVENT;
+        case "find":
+            return CommandType.FIND;
+        case "rm":
+            // Fallthrough
+        case "delete":
+            return CommandType.DELETE;
+        case "done":
+            return CommandType.DONE;
+        case "ls":
+            return CommandType.LIST;
+        default:
+            return CommandType.UNRECOGNISED;
+        }
+    }
+
+    public static void main(String[] args) {
+        System.out.println("Hello I'm Duke!");
```

Comment:
One suggestion (after looking at AB2's code) would be to make this into a method though the current one is fine as well!
eg. UserInterface.showGreetingMessage();